File: pytools/data/data.code.lua.server/MsgScenePlayer.lua
module("MsgScenePlayer", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = MsgScenePlayer})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.uid = 0
	self.scene_rot_pos = nil
end


function encode(self)
	pack = packet.new()
	pack:writeU32(self.uid)
	pack:writeBytes(self.scene_rot_pos:getBytes())
	return pack:readBytes()
end

function decode(self, pack)
	self.uid = pack:readU32()
	self.scene_rot_pos = MsgSceneRotPos.new():decode(pack)
	return self
end

function getBytes(self)
	return self:encode()
end


function setUid(self, uid)
	self.uid = uid
end
function getUid(self)
	return self.uid
end

function setSceneRotPos(self, scene_rot_pos)
	self.scene_rot_pos = scene_rot_pos
end
function getSceneRotPos(self)
	return self.scene_rot_pos
end

File: pytools/data/data.code.php.client/MsgScenePlayer.php
<?php
namespace protocol;

use network\Packet;


class MsgScenePlayer
{
	private $uid = 0;
	private $sceneRotPos = null;


	public function __construct(&$packet=null)
	{
		if ($packet)
		{
			$this->decode($packet);
		}
	}

	public function decode(&$packet)
	{
		$this->uid = $packet->readU32();
		$this->sceneRotPos = new MsgSceneRotPos($packet);
	}

	public function encode()
	{
		$packet = new Packet();
		$packet->writeU32($this->uid);
		$packet->writeBytes($this->sceneRotPos->getBytes());
		return $packet->readBytes();
	}

	public function getBytes()
	{
		return $this->encode();
	}


	public function setUid($uid)
	{
		$this->uid = $uid;
	}
	public function getUid()
	{
		return $this->uid;
	}

	public function setSceneRotPos($sceneRotPos)
	{
		$this->sceneRotPos = $sceneRotPos;
	}
	public function getSceneRotPos()
	{
		return $this->sceneRotPos;
	}

}
